### ISSUE TITLE
SDK: Set Payment's `destination` field as mandatory

### ISSUE THREAD

**Issue by hydra-yse** [breez/breez-sdk-liquid#481]

**Comment by hydra-yse** [comment]
cc @roeierez The way the `destination` field calculation currently works is:
1. Send/Receive swaps => Use the swap-related invoice
2. Chain swaps => Use the claim_address
3. Liquid payment => Use address/BIP21

We expect this data to be always available for local instances, but this may not be the case for real-time synced swaps (e.g. we find an onchain tx that is related to a swap, but the initiator never sends the payment details, or simply hasn't sent them yet). 

So what should we do in that case? Skip displaying the payment until we get the details? Use a sane default for the `destination` field, or keep it as optional?

**Comment by danielgranhao** [comment]
> So what should we do in that case? Skip displaying the payment until we get the details? Use a sane default for the destination field, or keep it as optional?

IMO, keeping it optional is better than using a placeholder. Integrators should set the placeholder themselves because they would most likely like to customize it or even translate it. Also, omitting a payment because we don't know the destination might be too extreme.

**Comment by roeierez** [comment]
> We expect this data to be always available for local instances, but this may not be the case for real-time synced swaps (e.g. we find an onchain tx that is related to a swap, but the initiator never sends the payment details, or simply hasn't sent them yet).

Not sure I am following the case here. If the swap doesn't exist (for example when the user doesn't activate realtime sync) we treat the payment as regular onchain payment (liquid payment) don't we?

**Comment by hydra-yse** [comment]
Yes, we do. Then it seems we can remove the optional value once the incoming chain swap case (pre-selecting the claim address) is resolved.
